integration_tests/src/main/python/delta_lake_optimize_table_test.py

```diff
@@ -45,20 +45,36 @@ def write_then_optimize(spark, path):
                               "ExecutedCommandExec")
 
 
-def _write_many_small_files(spark, enable_deletion_vectors, path, partition_columns=None):
+def _write_many_small_files(spark, enable_deletion_vectors, path, partition_columns=None, clustering_columns=None,
+                            tmp_view_name=None):
+    if partition_columns and clustering_columns:
+        raise ValueError("Only one of partition_columns or clustering_columns can be specified")
+
     num_slices = 64 if partition_columns is None else 4
     df = three_col_df(
         spark,
         SetValuesGen(IntegerType(), range(1000)),
         SetValuesGen(StringType(), list("abcdefghij")),
         string_gen,
         num_slices=num_slices)
-    writer = df.write.format("delta").mode("overwrite")
-    if partition_columns:
-        writer = writer.partitionBy(partition_columns)
-    if supports_delta_lake_deletion_vectors():
-        writer = writer.option("delta.enableDeletionVectors", str(enable_deletion_vectors).lower())
-    writer.save(path)
+    if clustering_columns:
+        assert tmp_view_name is not None, "A temp view name must be provided when using clustering"
```

Comment:
In Spark 3.4.0+ which looks like a given for these tests we can use Python formatted SQL to pass a dataframe directly without creating a temp view
https://github.com/apache/spark/blob/f78f164c5ea3275767a69d16dff80d22b052afd3/python/pyspark/sql/session.py#L1736